### PR TITLE
224 FEAT: Add function to get swap path by type

### PR DESCRIPTION
**Pull request by setrix422**

### CODE REVIEW

File: src/services/joeRouter.ts [hunk 1]
```diff
@@ -15,3 +19,24 @@ export const JoeRouter = (params: UpdateJoeRouterParams): Contract => {
...
+    toToken: state.tokens.tokens[state.swap[Field.OUTPUT]],
+  }));
+  const { fromTokenAddress, toTokenAddress } = {
+    fromTokenAddress: fromToken.address,
+    toTokenAddress: toToken.address,
+  };
+  const swapType = getSwapAvalancheType({ fromToken, toToken });
+
+  const swapPaths = {
+    [SwapAvalancheType.swapExactAVAXForTokens]: [fromTokenAddress, WAVAX, toTokenAddress],
```
Comment (AlinaYozh) [inline src/services/joeRouter.ts:36]: Incorrect type name - this type is correct where [WAVAX, tokenAddress] . Array with 3 addresses should have type **ExactTokenToToken**. 

That was described in the [task](https://github.com/yozh-io/1inch-widget/issues/224)

**Review by AlinaYozh — CHANGES_REQUESTED** [review]

File: src/services/joeRouter.ts (hunk 1 above)
Comment (setrix422) [reply src/services/joeRouter.ts:36]: So, to double check, in case ExactTokenToToken we need to return array with token instances?

Comment (AlinaYozh) [reply src/services/joeRouter.ts:36]: **The mistake is - type swapExactAVAXForTokens is used twice** 

In all cases need to return an array.

Joe router is modified uniswap v2 router, so we can use uniswap docs:

> Swaps an exact amount of input tokens for as many output tokens as possible, along the route determined by the path. The first element of path is the input token, the last is the output token, and any intermediate elements represent intermediate pairs to trade through (if, for example, a direct pair does not exist).

Links:

- [TokensForTokens](https://docs.uniswap.org/contracts/v2/reference/smart-contracts/router-02#swapexacttokensfortokens)
- [EthForTokens](https://docs.uniswap.org/contracts/v2/reference/smart-contracts/router-02#swapexactethfortokens)
- [TokensForEth](https://docs.uniswap.org/contracts/v2/reference/smart-contracts/router-02#swaptokensforexacteth)

Comment (setrix422) [reply src/services/joeRouter.ts:36]: Oh, sure, got it)

**Review by AlinaYozh — COMMENTED** [review]
Screenshot 2022-12-29 at 23 07 15

**Comment by setrix422** [comment]
Error with module absence will be fixed after pr on [https://github.com/yozh-io/1inch-widget/issues/222](issue) will be merged

**Review by Richardowsky — APPROVED** [review]

File: src/services/joeRouter.ts [hunk 2]
```diff
@@ -15,3 +19,24 @@ export const JoeRouter = (params: UpdateJoeRouterParams): Contract => {
 
   return contract;
 };
+
+export const getSwapPath = async (params: UpdateJoeRouterParams): Promise<any[]> => {
+  const WAVAX = await JoeRouter(params).WAVAX();
+  const { fromToken, toToken }: { fromToken: Token; toToken: Token } = useAppSelector((state) => ({
```
Comment (heimlee) [inline src/services/joeRouter.ts:25]: When calling getSwapPath function, there will be such error with useAppSelector hook:

image

Comment (setrix422) [reply src/services/joeRouter.ts]: Token instances have been moved  to method params

**Pull request merged**